ddtrace/_trace/trace_handlers.py

```diff
@@ -121,11 +122,22 @@ def _start_span(ctx: core.ExecutionContext, call_trace: bool = True, **kwargs) -
     distributed_context = ctx.get_item("distributed_context")
     if distributed_context and not call_trace:
         span_kwargs["child_of"] = distributed_context
+
+    # dispatch event for checking headers and possibly making an inferred proxy span
+    core.dispatch("inferred_proxy.start", (ctx, tracer, span_kwargs, call_trace, distributed_headers_config))
+    # re-get span_kwargs in case an inferred span was created and we have a new span_kwargs.child_of field
+    span_kwargs = ctx.get_item("span_kwargs")
+
     span_kwargs.update(kwargs)
     span = (tracer.trace if call_trace else tracer.start_span)(ctx["span_name"], **span_kwargs)
```

Comment:
Just adding a note to fix this: Even if the above line was fixed: `span_kwargs = ctx.get_item("span_kwargs")`, when it hits this line, it throws an error about tracer.trace not accepting a child of argument:
>TypeError: Tracer.trace() got an unexpected keyword argument 'child_of'

The reason it worked in https://github.com/DataDog/dd-trace-py/pull/11938/files is because it used start_span which supports it.

Reply:
this should be fixed, there has to be a check to ensure `trace()` is not being called since it auto-handles context.